staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -88,6 +90,14 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 		Dial: c.Dial,
 	}
 
+	if c.ProxyURL != "" {
```

Comment:
The doc says http, https, and socks5 are supported, and the scheme defaults to http. Where is that defaulting/validation done?

Reply:
Added validation, removed defaulting.